### DISCUSSION TITLE
The new on property improves output testing in Angular Testing Library

### DISCUSSION THREAD

**Discussion by giscus** [timdeschryver/timdeschryver.dev#352] [Announcements]
# The new on property improves output testing in Angular Testing Library

A quick look at the new `on` API to write output tests more intuitively and without workarounds.

https://timdeschryver.dev/blog/the-new-on-property-improves-output-testing-in-angular-testing-library

**Comment by peterhexakia** [comment]
With componentProperties being deprecated in the future, how can we override a protected property in a test?
For example, a component has a protected property images (it is not an input).

Currently, I can do the following to mock the images in the test so I can test its behavior. Is there a way to do this without using componentProperties?
```
componentProperties: {
   images: mockImages,
}
```

**Reply by timdeschryver** [reply]
I think `componentProperties` will need to be undeprecated to handle "edge" cases.
This is one of them.
Another example is to test inputs via a template, for more info see https://github.com/testing-library/angular-testing-library/issues/470
